Kernel/Storage/ATA/AHCIPortHandler.cpp

```diff
@@ -89,10 +45,8 @@ bool AHCIPortHandler::handle_irq(RegisterState const&)
     if (m_pending_ports_interrupts.is_zeroed())
         return false;
     for (auto port_index : m_pending_ports_interrupts.to_vector()) {
-        auto port = m_handled_ports[port_index];
-        VERIFY(port);
         dbgln_if(AHCI_DEBUG, "AHCI Port Handler: Handling IRQ for port {}", port_index);
-        port->handle_interrupt();
+        m_parent_controller->handle_interrupt_for_port({}, port_index);
```

Comment:
This way we can put a IRQ handler where we want it to be, so if we support IOAPIC, it might require us to put it in a different IRQ vector. MSI and MSI-X interrupts require multiple handlers, so this achieves that quite well.